client/src/modules/CourseManagement/components/CourseTaskModal/index.tsx

```diff
@@ -136,40 +144,58 @@ export function CourseTaskModal(props: Props) {
           </Form.Item>
         </Col>
       </Row>
-
       {changes?.checker === 'crossCheck' ? (
-        <Form.Item name="pairsCount" label="Cross-Check Pairs Count">
-          <Select placeholder="Cross-Check Pairs Count">
-            {times(10, num => (
-              <Option key={num} value={num + 1}>
-                {num + 1}
-              </Option>
-            ))}
-          </Select>
-        </Form.Item>
+        <>
+          <Divider style={{ marginTop: 0, marginBottom: 8 }} />
+          <Typography.Title level={4}>Cross-Check</Typography.Title>
+          <Row gutter={24}>
+            <Col span={12}>
+              <Form.Item name="crossCheckEndDate" label="Cross-Check End Date">
+                <DatePicker format="YYYY-MM-DD" />
+              </Form.Item>
+            </Col>
+            <Col span={12}>
+              <Form.Item name="pairsCount" label="Cross-Check Pairs Count">
+                <Select placeholder="Cross-Check Pairs Count">
+                  {times(10, num => (
+                    <Option key={num} value={num + 1}>
+                      {num + 1}
```

Comment:
It seems that `num + 1` is duplicated here, perhaps, we could extract it in variable

Reply:
as it's something that was there and since it will make code for verbose inside tsx, let me ignore it now :)